shelljs/shelljs.d.ts

```diff
@@ -434,29 +434,42 @@ declare module "shelljs"
      * Object containing environment variables (both getter and setter). Shortcut to process.env.
      */
     export var env: { [key: string]: string };
-
-    /*
-
-    // Not yet implemented due to implementation issues (constant overloads and return types).
-    // See: https://github.com/arturadib/shelljs#execcommand--options--callback
-
-    export function exec(command: string, options: ExecOptions, callback: (code: number, output: string) => any): any;
-    export function exec(command: string, options: ExecOptions): any;
-
-    interface ExecOptions
-    {
-        silent: boolean;
-        async: boolean;
-    }
-
-    */
-
+    
     /**
      * Executes the given command synchronously.
-     * @param  {string}          command The commadn to execute.
+     * @param  {string}          command The command to execute.
      * @return {ExecReturnValue}         Returns an object containing the return code and output as string.
      */
     export function exec(command: string): ExecReturnValue;
+    /**
+     * Executes the given command synchronously.
+     * @param  {string}          command The command to execute.
+     * @param  {ExecOptions}     options Silence and synchronous options.
+     * @return {ExecReturnValue}         Returns an object containing the return code and output as string.
+     */
+    export function exec(command: string, options: ExecOptions): ExecReturnValue;
```

Comment:
There were issues implementeing that function because the return type depends on the `async` option provided.
See docs:
> When in synchronous mode returns the object { code:..., output:... }, containing the program's output (stdout + stderr) and its exit code. Otherwise returns the child process object, and the callback gets the arguments (code, output).

This function actually has two return types, which is not possible to model through a constant overload (overloads must have the same return types).
There are basically two solutions for this IMO:
1. Make `ExecReturnValue` inherit from ChildProcess. I wouldn't do that because it may be misleading.
2. Since there are Union types nor available, this may be a solution:
`export function exec(command: string, options: ExecOptions): ExecReturnValue|ChildProcess;`

If I didn't get something right, please tell me.